runtime/src/main/java/org/corfudb/runtime/LogReplicationUtils.java

```diff
@@ -86,6 +86,50 @@ public static void subscribe(@Nonnull LogReplicationListener clientListener, @No
         log.info("Client subscription at timestamp {} successful.", subscriptionTimestamp);
     }
 
+    public static void subscribeRqListener(@Nonnull LogReplicationRoutingQueueListener clientListener, @Nonnull String namespace,
+                                           int bufferSize, CorfuStore corfuStore) {
+
+        long subscriptionTimestamp = getRoutingQueueSubscriptionTimestamp(corfuStore, namespace, clientListener);
+        // Open the routing queue from corfu store (Disk backed mode)
+        // since the subscribe API will run in policy jvm
+        // TODO: Get client name as input params. For now, hard coding the stream tag.
+        if (checkIfRoutingQueueExists(corfuStore, namespace, REPLICATED_QUEUE_TAG_PREFIX)) {
+            // Table registry contains the routing queue already.
+            corfuStore.getRuntime().getTableRegistry().getStreamingManager().subscribeLogReplicationRoutingQueueListener(
+                    clientListener, namespace, subscriptionTimestamp, bufferSize,
+                    getRoutingQueue(corfuStore, namespace, REPLICATED_QUEUE_TAG_PREFIX));
+            log.info("Routing queue client subscription at timestamp {} successful.", subscriptionTimestamp);
+        } else {
+            // Routing queue is not registered at the sink (receiver side) yet.
+            // Subscribe to LR status table and poll for routing queue registry
+            corfuStore.getRuntime().getTableRegistry().getStreamingManager().subscribeLogReplicationLrStatusTableListener(
+                    clientListener, subscriptionTimestamp, bufferSize);
+            log.info("Subscribed to LR status table at timestamp {}.", subscriptionTimestamp);
+        }
+    }
+
+    public static void subscribeRqListenerWithTs(@Nonnull LogReplicationRoutingQueueListener clientListener,
+                                                 @Nonnull String namespace, int bufferSize, CorfuStore corfuStore,
+                                                 long subscriptionTimestamp) {
+
+        // Open the routing queue from corfu store (Disk backed mode)
+        // since the subscribe API will run in policy jvm
+        // TODO: Get client name as input params. For now, hard coding the stream tag.
+        if (checkIfRoutingQueueExists(corfuStore, namespace, REPLICATED_QUEUE_TAG_PREFIX)) {
+            // Table registry contains the routing queue already.
+            String routingQueueName = getRoutingQueue(corfuStore, namespace, REPLICATED_QUEUE_TAG_PREFIX);
+            corfuStore.getRuntime().getTableRegistry().getStreamingManager().subscribeLogReplicationRoutingQueueListener(
+                    clientListener, namespace, subscriptionTimestamp, bufferSize, routingQueueName);
+            log.info("Routing queue client subscription at timestamp {} successful.", subscriptionTimestamp);
+        } else {
```

Comment:
we don't need to subscribe to the replication status table now.
So in the else part, we can just keep checking if a record for routing table exists in the registry table.
For PoC we can go with just a scheduled polling.

Reply:
+1

Reply:
Yes, let me add a todo here to discover the queue without listening on status table.